### PR TITLE
fix private module registry sources

### LINKED ISSUE #302: Modules from Private Registries can not be parsed
**Describe the bug**
Modules from private registries can not be parsed and will be ignored. yor outputs a warning and ignores the module.

**To Reproduce**
1. Use yor in terraform stack where at least one module is referenced from a private module registry
2. Make sure to have valid credentials for private registry `terraform login some.private.registry`
3. Run `yor tag -d .` in terraform stack
4. Yor ignores module and outputs following warning
```
[WARNING] failed to parse hcl module in directory some.private.registry/namespace/name/aws/some.private.registry/namespace/name/aws because of errors Failed to read module directory: Module directory some.private.registry/namespace/name/aws does not exist or cannot be read.
```

**Expected behavior**
Yor parses and tags module.

**Desktop:**
 - OS: MacOS
 - Yor Version: 0.1.151

**Additional context**
```hcl
module "example" {
  source  = "some.private.registry/namespace/name/aws"
  version = "1.0.0"
}
```

### PR DESCRIPTION
**Pull request by mhennecke**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

fix #302

Generalizes PR #276

### CODE REVIEW

**Comment by mhennecke** [comment]
@nimrodkor @tsmithv11 @rotemavni Workflow approval and review is very much appreciated

File: src/terraform/structure/terraform_module_test.go [hunk 1]
```diff
@@ -16,11 +16,6 @@ func TestTerrraformModule(t *testing.T) {
 		assert.True(t, isRemote)
 	})
 
-	t.Run("Test TF Module remote scalr logic", func(t *testing.T) {
-		isRemote := isRemoteModule("jameswoolfenden.scalr.io/acc-u1ksa0vgdflusgo/cloudfront/aws")
-		assert.True(t, isRemote)
-	})
```
Comment (nimrodkor) [inline src/terraform/structure/terraform_module_test.go:22]: @mhennecke Why is this test not needed anymore? AFAIK, this should still work, right?

**Review by nimrodkor — COMMENTED** [review]
Thanks @mhennecke !

File: src/terraform/structure/terraform_module_test.go (hunk 1 above)
Comment (mhennecke) [reply src/terraform/structure/terraform_module_test.go:22]: @nimrodkor It is still there. I just moved it a bit further down in the file.

**Review by nimrodkor — APPROVED** [review]
Perfect, TY!

**Pull request merged**